src/tests/ftest/util/job_manager_utils.py

```diff
@@ -60,19 +63,56 @@ def get_job_manager(test, class_name=None, job=None, subprocess=None, mpi_type=N
 
     # Setup a job manager command for running the test command
     if class_name is not None:
-        job_manager = get_job_manager_class(class_name, job, subprocess, mpi_type)
-        job_manager.get_params(test)
-        job_manager.timeout = timeout
-        if mpi_type == "openmpi" and hasattr(job_manager, "tmpdir_base"):
-            job_manager.tmpdir_base.update(test.test_dir, "tmpdir_base")
-        if isinstance(test.job_manager, list):
-            test.job_manager.append(job_manager)
-        else:
-            test.job_manager = job_manager
+        job_manager = add_job_manager(test, class_name, job, subprocess, mpi_type, timeout)
 
     return job_manager
 
 
+def add_job_manager(test, class_name, job, subprocess, mpi_type, timeout):
```

Comment:
Why have two methods for getting a job manager?

Reply:
The original thought was to move to a more direct method than the existing method, but I have changed my mind.